src/py_profet/interactive_plots/new_dash.py

```diff
@@ -107,12 +107,35 @@ def save_pdf(trace_file: str):
     # print(dict(system_arch))
     
     # allow a maximum of elements to display. Randomly undersample if there are more elements than the limit
+    '''
+    # Ordering by timestamp after undersampling
+    max_elements = 10000
+    if len(df) > max_elements:
+        df = df.sort_values(by='stress_score')
+        k = len(df) // max_elements
+        indices_to_select = np.arange(0, len(df), k)
+
+        sampled_df = df.iloc[indices_to_select].copy()
+        sampled_df = sampled_df.append(df.nsmallest(3, 'stress_score'))
+        sampled_df = sampled_df.append(df.nlargest(3, 'stress_score'))
+        df = sampled_df
+        df = df.sort_values(by='timestamp')
+    else:
+        max_elements = None
+    '''
+
+    # Undersampling based on stress_score
     max_elements = 10000
     if len(df) > max_elements:
-        # TODO make each socket have the same number of elements
```

Comment:
This TODO is very important. Let's implement this already:

The proposed sampling is not taking into account the different sockets, so if there is a big unbalance between sockets (and it can easily happen) then we're completely biasing our visualizations, as we're plotting everything per socket.

My proposal is to perform this sampling per socket (uniquely identified by node and socket number). However, the `max_elements` variable should apply to all the charts together, meaning that the number of points to plot per socket is `max_elements / num. sockets` (e.g. with 2 nodes and 2 sockets per node, it'd be 10,000 / 4 = 2,500 data points per socket if  `max_elements = 10000`).

Reply:
Check out what I did and see what you think! I believe this is what you meant, right? 

Once we know the data per socket (in your example, the 2500) we group the df by socket and see if the socket has more than 2500 elements, if it does we sample it using the systematic sampling technique, if it doesn't we simply save all the points it has.

I believe this is what you meant.

Reply:
Clear! Makes sense. Checkout the new commit to see if I did what you meant.